aspnetcore/mvc/controllers/application-model.md

```diff
@@ -12,7 +12,187 @@ ms.technology: aspnet
+* [`CorsApplicationModelProvider`](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.cors.internal.corsapplicationmodelprovider)
 
 > [!NOTE]
-> We are currently working on this topic.We welcome your input to help shape the scope and approach. You can track the status and provide input on this [issue](https://github.com/aspnet/Docs/issues/138) at GitHub.If you would like to review early drafts and outlines of this topic, please leave a note with your contact information in the [issue](https://github.com/aspnet/Docs/issues/138).Learn more about how you can [contribute](https://github.com/aspnet/Docs/blob/master/CONTRIBUTING.md) on GitHub.
+> The order in which two providers with the same value for `Order` are called is undefined, and therefore should not be relied upon.
+
+The `DefaultApplicationModelProvider` establishes many of the default behaviors used by ASP.NET Core MVC. Its responsibilities include:
+
+* Adding global filters to the context
+* Adding controllers to the context
+* Adding public controller methods as actions
+* Adding action method parameters to the context
+* Applying route and other attributes
+
+Some built-in behaviors are implemented by the `DefaultApplicationModelProvider` and can be overridden by replacing its behavior. For example, normally static methods are not added as controller actions. You can change this by overriding the virtual `IsAction` method. This provider is also responsible for constructing the [`ControllerModel`](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.applicationmodels.controllermodel), which in turn references [`ActionModel`](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.applicationmodels.actionmodel#Microsoft_AspNetCore_Mvc_ApplicationModels_ActionModel), [`PropertyModel`](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.applicationmodels.propertymodel), and [`ParameterModel`](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.applicationmodels.parametermodel#Microsoft_AspNetCore_Mvc_ApplicationModels_ParameterModel) instances.
+
+The `AuthorizationApplicationModelProvider` is responsible for applying the behavior associated with the `AuthorizeFilter` and `AllowAnonymousFilter` attributes. [Learn more about these attributes](https://docs.microsoft.com/aspnet/core/security/authorization/simple).
+
+The `CorsApplicationModelProvider` implements behavior associated with the `IEnableCorsAttribute` and `IDisableCorsAttribute`, and the `DisableCorsAuthorizationFilter`. [Learn more about CORS](https://docs.microsoft.com/aspnet/core/security/cors).
+
+## Conventions
+
+The application model defines convention abstractions that provide a simpler way to customize the behavior of the models than overriding the entire model or provider. These abstractions are the recommended way to modify your app's behavior.
```

Comment:
> These abstractions are the recommended way to modify your app's behavior.

👍 

I would mention here that conventions are the way for you to write **code that will dynamically apply customizations**. You can modify the framework's behavior already by defining filters and such, application model lets you control how things are wired together. If attributes made you sad, we have an antidote!